pkg/server/options/serving.go

```diff
@@ -307,7 +307,7 @@ func (s *SecureServingOptions) MaybeDefaultWithSelfSignedCerts(publicAddress str
 		}
 	}
 
-	if !canReadCertAndKey {
+	if !canReadCertAndKey && s.ServerCert.GeneratedCert == nil {
```

Comment:
hmm...the current API of MaybeDefaultWithSelfSignedCerts() is that it takes a public address as input, and populates GeneratedCert with a cert for that address.

this changes the contract so that, now, GeneratedCert is both an input and an output? and if it's used as an input, it ignores the other inputs? that feels pretty weird to me.

what if ServingOptions had a field like `PregeneratedCert`, and skipped the whole "MaybeDefaultWithSelfSignedCerts" function call if that field exists?

Reply:
I was a bit torn on that behavior; I've added the new boolean and a check in MaybeDefaultWithSelfSignedCerts to exit early if set.